### PR TITLE
feat(frontend): service to batch promises

### PR DESCRIPTION
**Pull request by AntonioVentilii**
# Motivation

Splitting PR #3644 , we create first a services useful to break a list of promises in smaller batches.

# Tests

Test provided.

### CODE REVIEW

File: src/frontend/src/lib/services/batch.services.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+export const batchPromisesPerSecond = async function* <T, R>({
```
Comment (peterpeterparker) [inline src/frontend/src/lib/services/batch.services.ts:1]: Maybe just `batchPromises` or `batch`? I like `batch` actually wdyt?

File: src/frontend/src/lib/services/batch.services.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+	items,
+	maxCallsPerSecond,
+	promise,
+	validate,
+	afterPromise
+}: {
+	items: T[];
+	maxCallsPerSecond: number;
```
Comment (peterpeterparker) [inline src/frontend/src/lib/services/batch.services.ts:9]: I did not understood `maxCallsPerSecond` from the naming. Now that I read the code I get it but, it was not natural. Any other naming idea?

**Review by peterpeterparker — COMMENTED** [review]
What about simplifying and just passing all the promises?

Also not sure what's the use case for `afterPromise`?

File: src/frontend/src/lib/services/batch.services.ts (hunk 1 above)
Comment (AntonioVentilii) [reply src/frontend/src/lib/services/batch.services.ts:1]: batch!

File: src/frontend/src/lib/services/batch.services.ts (hunk 2 above)
Comment (AntonioVentilii) [reply src/frontend/src/lib/services/batch.services.ts:6]: I went with `batchSize` ? WDYT

**Comment by AntonioVentilii** [comment]
@peterpeterparker as discussed offline, I simplified the logic and now it takes just a list of promises as input

File: src/frontend/src/lib/services/batch.services.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+export const batch = async function* <T>({
+	promises,
+	batchSize
```
Comment (peterpeterparker) [inline src/frontend/src/lib/services/batch.services.ts:3]: Just `size`? Don't know, today I'm in the mood of shortening any variable names 😂

**Review by peterpeterparker — APPROVED** [review]
LGTm, thx

File: src/frontend/src/lib/services/batch.services.ts (hunk 3 above)
Comment (AntonioVentilii) [reply src/frontend/src/lib/services/batch.services.ts:3]: ahahahahah! I noticed! I am just afraid that it is not clear enough, it should say "how many promises per batch"

**Comment by AntonioVentilii** [comment]
[OISY-130]

**Pull request merged**